lib/DelayedEventAPI.php

```diff
@@ -5,255 +5,118 @@
 require 'Error.php';
 
 /**
- * TTG Delayed Event PHP Client
- * @author waffles
+ * TTG New Delayed Event PHP Client
+ * @author slobo
  */
 
 class DelayedEventAPI
 {
 	const HTTP_POST = 'POST';
 	const HTTP_GET = 'GET';
-	const HTTP_PUT = 'PUT';
-	const HTTP_DELETE = 'DELETE';
 
-	protected $username;
-	protected $password;
 	protected $api_host;
+	protected $callback_uri;
+	protected $parameters;
```

Comment:
The parameters, fire_date, and callback_uri should be sent in to the create() method. Only the api_host will be needed in the constructor. Also, the fire_events() method is not necessary. The events are fired from a lambda function automatically.